src/v/config/property.h

```diff
@@ -362,6 +374,10 @@ class binding_base {
         _hook.swap_nodes(rhs._hook);
     }
 
+    const T& do_maybe_sanctioned_value(bool should_sanction) const {
+        return _parent->maybe_sanctioned_value(should_sanction);
+    }
+
```

Comment:
Something that I really don't like is how this circumvents the cached value in the binding altogether and it goes straight to the parent to ask for the sanction logic. To make things worse, they both return a reference, but to a different values. One is the local copy and the other is the property value. If this were to lead to a bug, it is the type that is a very big pain to debug.

However, I am not sure on how to expose this without either copying the whole `restricted` logic, including the default value, from the `enterprise` property. The problem is that this is a dynamic check, while the binding itself is a way to cache the latest value and not talk to the parent on the call site.

Thoughts?

Reply:
> Thoughts?

Is a new interface required? I.e., will both the original and sanctioned value ever be required from the same `binding`?

It may be useful to have a `sanctioned_binding` that caches the sanctioned value.

Reply:
> I.e., will both the original and sanctioned value ever be required from the same binding?

We don't currently have a case where we need both. I would assume, no.

> It may be useful to have a `sanctioned_binding`

I could split them up, even for just separation of concern :+1: 

> that caches the sanctioned value.

I was going to reply "this might be tricky". But while i was writing the reasons, i think i found a simple way to do it. I will give it a go.